clang/test/CodeGenSYCL/int_header1.cpp

```diff
@@ -76,22 +76,6 @@ struct MyWrapper {
     // translation unit scope
     kernel_single_task<nm1::nm2::KernelName0>([=]() { acc.use(); });
 
-#ifdef LI__
-    // TODO unexpected compilation error when host code + integration header
-    // is compiled LI-- kernel name is an incomplete class forward-declared in
-    // local scope
-    class KernelName2;
-    kernel_single_task<KernelName2>([=]() { acc.use(); });
-#endif
-
-#ifdef LD__
```

Comment:
I suggest to convert these to negative compilation tests together with removal. Or leave it as is if there are no resource now.

Reply:
I preserved these negative test cases in the new test, this is mentioned in commit message